Make noarch

m-rossi opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

As discussed in https://github.com/pythonnet/pythonnet/discussions/1864 package can be `noarch`

m-rossi committed:
Make package noarch

m-rossi committed:
Remove dotnet variants and test only against dotnet-runtime 6.0

m-rossi committed:
Bump build number

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.
 * noarch: python recipes are required to have a lower bound on the python version. Typically this means putting `python >=3.6` in **both** `host` and `run` but you should check upstream for the package's Python compatibility.

m-rossi commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.21.9, conda-smithy 3.21.0, and conda-forge-pinning 2022.07.11.00.47.13

m-rossi committed:
Require Python 3

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.

m-rossi committed:
Remove mono as runtime requirement

https://github.com/pythonnet/pythonnet/discussions/1864#discussioncommen
t-3120826

m-rossi committed:
Remove requirements related to specific centos image

m-rossi committed:
Install mono during test

m-rossi commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.21.9, conda-smithy 3.21.0, and conda-forge-pinning 2022.07.11.12.35.27

m-rossi committed:
Add missing import

m-rossi committed:
Try to remove the compiler requirement

Leftover for centos 7

m-rossi committed:
Remove selector for PyPy

Meta-package for cffi on PyPy is available

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.
